kv/kvserver: TestReplicateQueueDecommissioningNonVoters failed

kv/kvserver.TestReplicateQueueDecommissioningNonVoters [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9185169?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9185169?buildTab=artifacts#/) on release-23.1 @ [80c4895c566a7eaa6f16c4098980509dd3795ad7](https://github.com/cockroachdb/cockroach/commits/80c4895c566a7eaa6f16c4098980509dd3795ad7):


```
=== RUN   TestReplicateQueueDecommissioningNonVoters
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters2486490881
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestReplicateQueueDecommissioningNonVoters
    replicate_queue_test.go:819: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters2486490881
--- FAIL: TestReplicateQueueDecommissioningNonVoters (109.23s)
=== RUN   TestReplicateQueueDecommissioningNonVoters/remove
    replicate_queue_test.go:793: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/replicate_queue_test.go:793
        	Error:      	Condition never satisfied
        	Test:       	TestReplicateQueueDecommissioningNonVoters/remove
    --- FAIL: TestReplicateQueueDecommissioningNonVoters/remove (68.62s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueDecommissioningNonVoters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-25756

--- a/pkg/kv/kvserver/replicate_queue_test.go
+++ b/pkg/kv/kvserver/replicate_queue_test.go
@@ -622,6 +622,7 @@ func TestReplicateQueueDecommissioningNonVoters(t *testing.T) {
 	defer log.Scope(t).Close(t)
 	skip.UnderRace(t, "takes a long time or times out under race")
 	skip.UnderDeadlockWithIssue(t, 94383)
+	skip.UnderMetamorphicWithIssue(t, 99207)
 
 	ctx := context.Background()
 

--- a/pkg/testutils/skip/skip.go
+++ b/pkg/testutils/skip/skip.go
@@ -148,6 +148,18 @@ func UnderMetamorphic(t SkippableTest, args ...interface{}) {
 	}
 }
 
+// UnderMetamorphicWithIssue skips this test during metamorphic runs, which are
+// tests run with the metamorphic build tag, logging the given issue ID as the
+// reason.
+func UnderMetamorphicWithIssue(t SkippableTest, githubIssueID int, args ...interface{}) {
+	t.Helper()
+	if util.IsMetamorphicBuild() {
+		t.Skip(append([]interface{}{fmt.Sprintf(
+			"disabled under metamorphic. issue: https://github.com/cockroachdb/cockroach/issues/%d", githubIssueID,
+		)}, args...))
+	}
+}
+
 // UnderNonTestBuild skips this test if the build does not have the crdb_test
 // tag.
 func UnderNonTestBuild(t SkippableTest) {

